src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -658,6 +661,84 @@ TEST_F(ControllerScriptEngineLegacyTest, connectionExecutesWithCorrectThisObject
     EXPECT_DOUBLE_EQ(1.0, pass->get());
 }
 
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetUndefinedOnUnknownCharset) {
+    const auto result = evaluate("engine.convertCharset('NULL', 'Hello!')");
+
+    EXPECT_EQ(qjsvalue_cast<QByteArray>(result), QByteArrayView(""));
+}
```

Comment:
[Ditto](https://github.com/mixxxdj/mixxx/pull/13935/files#r1862222749).

Reply:
yeah... Not quite sure what to do here honestly. It seems that there are implicit conversions to enums happening, which isn't great if it just silently succeeds. I wonder what value plain `undefined` results in? I'd guess `undefined`->`0`->`US_ASCII`?

Reply:
I just removed it for now. I don't really know how what the proper way to handle implicit enum conversions would be. We could take a QJSValue instead and implement our own checks, but ideally Qt would warn by itself.